lib/route_consistency_checker.rb

```diff
@@ -0,0 +1,145 @@
+require "set"
+
+require "gds_api/router"
+
+class RouteConsistencyChecker
+  attr_reader :errors
+
+  def initialize(routes, router_data)
+    @routes = load_routes(routes)
+    @remaining_routes = Set.new(@routes.keys)
+    @errors = Hash.new { |hash, key| hash[key] = [] }
+    @router_data = load_router_data(router_data)
+  end
+
+  def check_content
+    content_items_to_check.each do |content_item|
+      content_item.redirects.each do |redirect|
+        check_redirect(content_item, redirect)
+      end
+
+      content_item.routes.each do |route|
+        check_route(content_item, route)
+      end
+    end
+  end
+
+  def check_routes
+    @remaining_routes.each do |path|
+      next if @router_data.include?(path)
+
+      route = @routes.fetch(path)
+      next if route["disabled"]
+      next if route["handler"] == "gone"
+      next if route["handler"] == "redirect" && !route["backend_id"].present?
+
+      content_item = ContentItem.find_by_path(route["incoming_path"])
```

Comment:
we want to look up the content item by an exact route match here - I don't think we're worried about base_path